In [2]:
import pandas as pd
import numpy as np

In [3]:
df =pd.read_csv("../data/backtest_results/monthly_performance.csv")


cummulative_return=(df["portfolio_value"].iloc[-1]/df["portfolio_value"].iloc[0])-1

#monthly stats
mean_return= df["return_pct"].mean()
volatility_monthly= df["return_pct"].std()

#annualized metrics, standard formula monthly standard deviation times sqrt(12)
volatility_annual = volatility_monthly*np.sqrt(12)
sharpe_ratio = mean_return/volatility_monthly if volatility_monthly!=0 else np.nan

df["cum_max"] = df["portfolio_value"].cummax()
df["drawdown"] = df["portfolio_value"]/df["cum_max"]-1
max_drawdown = df["drawdown"].min()
calmar_ratio = cummulative_return/abs(max_drawdown) if max_drawdown != 0 else np.nan

summary = {
    "Cummulatice return(%)": f"{cummulative_return:.2%}",
    "Monthly Volatility (%)": f"{volatility_monthly:.2%}",
    "Annulized Volatility(%)": f"{volatility_annual:.2%}",
    "Sharpe ratio": round(sharpe_ratio,2),
    "Max Drawdown(%)": f"{max_drawdown:.2%}",
    "Calmar Ratio": round(calmar_ratio,2),
}

summary_df = pd.DataFrame(summary.items(), columns=["Metrics","Value"])
summary_df.to_csv("../data/summary_metrics.csv")
summary_df

Metrics      Value
0    Cummulatice return(%)      7.40%
1   Monthly Volatility (%)   5146.96%
2  Annulized Volatility(%)  17829.58%
3             Sharpe ratio       0.81
4          Max Drawdown(%)     -3.78%
5             Calmar Ratio       1.96

## Summary & Insights

This project backtested a long-short momentum strategy across 2024.

- Portfolio started with $1,000.
- Monthly rebalancing based on top/bottom 20 momentum.
- Strategy metrics:
    - Sharpe Ratio: 0.81
    - Max Drawdown: -3.78%
    - Calmar Ratio: 1.96
    - Cumulative Return: 7.40%

**Visualizations are saved in the `visuals/` folder.**